ehr/resources/web/ehr/panel/BulkEditPanel.js

```diff
@@ -80,7 +80,10 @@ Ext4.define('EHR.panel.BulkEditPanel', {
                 }, this);
 
                 var keys = Ext4.Object.getKeys(values);
-                if (keys.length == 1){
+
+                // there can be more than one value for certain fields, ex. Date field with the time component,
+                // there might be different times in the selected rows, so pick one as a default value
+                if (keys.length >= 1){
```

Comment:
This doesn't feel quite right but maybe I'm not following it correctly. It seems like it will populate the Bulk Edit dialog with a value for *any* field that has at least one value, even though they're not the same. So if you have a field where rows have values 2, 5, and 9, it'll just choose one of them to populate the dialog?

The intent of this code is to have the dialog pop up with prepopulated with any values that are identical across the rows, and leave the rest blank. Is there a problem when none of the fields have matching values across all of the rows?

Reply:
This is definitely not how this is designed to work. What is breaking for WNPRC?

Reply:
@labkey-jeckels 
"So if you have a field where rows have values 2, 5, and 9, it'll just choose one of them to populate the dialog?"
With my change, that is correct.

"Is there a problem when none of the fields have matching values across all of the rows?"
Yes, its the same issue where the Bulk Edit window collapses.

@bbimber This is an attempt to fix this issue: https://teamcity.labkey.org/viewLog.html?buildId=2389394&buildTypeId=LabKey_227Release_Premium_Ehr_WnprcEhrPostgres

Reply:
@labkey-bpatel: OK. I get there is a JS error happening. When you repro locally, what's the actual exception? 

The way this component is supposed to work is that if you have 3 rows selected, it should find all instances where 100% of the rows have the same value. For any field where this is true, it should pre-populate the field. 

I could imagine there are maybe some type-conversion situations that could break it? It has definitely been used on situations involving date, date-time, and similar types of values before.

Reply:
If a field across selected rows have different values, then we can change it such that we do not allow Bulk Edit in the first place. Thoughts?

Reply:
@bbimber 
client side exception from chrome:
 at constructor.finishRender (ext-all-sandbox-debug.js?980774510:28219:42)
    at constructor.finishRenderItems (ext-all-sandbox-debug.js?980774510:35320:22)
    at constructor.finishRender (ext-all-sandbox-debug.js?980774510:35675:12)
    at constructor.finishRenderChildren (ext-all-sandbox-debug.js?980774510:37892:20)
    at constructor.afterRender (ext-all-sandbox-debug.js?980774510:27968:12)
    at constructor.callParent (ext-all-sandbox-debug.js?980774510:4272:32)
    at constructor.afterRender (ext-patches.js?980774510:136:16)
    at constructor.callParent (ext-all-sandbox-debug.js?980774510:4272:32)
    at constructor.afterRender (ext-all-sandbox-debug.js?980774510:33625:12)
    at constructor.finishRender (ext-all-sandbox-debug.js?980774510:28246:12)
finishRender @ ext-all-sandbox-debug.js?980774510:28219
finishRenderItems @ ext-all-sandbox-debug.js?980774510:35320
finishRender @ ext-all-sandbox-debug.js?980774510:35675
finishRenderChildren @ ext-all-sandbox-debug.js?980774510:37892
afterRender @ ext-all-sandbox-debug.js?980774510:27968
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-patches.js?980774510:136
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-all-sandbox-debug.js?980774510:33625
finishRender @ ext-all-sandbox-debug.js?980774510:28246
finishRenderItems @ ext-all-sandbox-debug.js?980774510:35320
finishRender @ ext-all-sandbox-debug.js?980774510:35675
finishRenderChildren @ ext-all-sandbox-debug.js?980774510:37892
afterRender @ ext-all-sandbox-debug.js?980774510:27968
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-patches.js?980774510:136
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-all-sandbox-debug.js?980774510:33625
finishRender @ ext-all-sandbox-debug.js?980774510:28246
finishRenderItems @ ext-all-sandbox-debug.js?980774510:35320
finishRender @ ext-all-sandbox-debug.js?980774510:35675
finishRenderChildren @ ext-all-sandbox-debug.js?980774510:37892
afterRender @ ext-all-sandbox-debug.js?980774510:27968
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-patches.js?980774510:136
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-all-sandbox-debug.js?980774510:33625
finishRender @ ext-all-sandbox-debug.js?980774510:28246
finishRenderItems @ ext-all-sandbox-debug.js?980774510:35320
finishRender @ ext-all-sandbox-debug.js?980774510:35675
finishRenderChildren @ ext-all-sandbox-debug.js?980774510:37892
afterRender @ ext-all-sandbox-debug.js?980774510:27968
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-patches.js?980774510:136
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-all-sandbox-debug.js?980774510:33625
callParent @ ext-all-sandbox-debug.js?980774510:4272
afterRender @ ext-all-sandbox-debug.js?980774510:85176
finishRender @ ext-all-sandbox-debug.js?980774510:28246
render @ ext-all-sandbox-debug.js?980774510:28512
doAutoRender @ ext-all-sandbox-debug.js?980774510:28118
show @ ext-all-sandbox-debug.js?980774510:34016
handler @ BulkEditWindow.js?980774510:96
callback @ ext-all-sandbox-debug.js?980774510:7250
onClick @ ext-all-sandbox-debug.js?980774510:123187
onClick @ ext-all-sandbox-debug.js?980774510:124053
eval @ VM1411:6
wrap @ ext-all-sandbox-debug.js?980774510:10572

It looks like even if i use plain and simple values across rows (screenshot below), its the same issue:
![image](https://user-images.githubusercontent.com/10765266/231903338-fd21ad92-1ea8-475c-ab2a-0367eb5b4f33.png)

Reply:
I believe that would defeat the purpose of this.

Do you have a local repro, and what specific js error is happening?

That behavior where an extjs window renders to the side often happens when an xtype isn't loaded

Reply:
Yeah seems like it.

Yes, I have a local report. Above is the only js error I am seeing.

Hmm, if xtype isn't getting loaded then I am not seeing an error related to that, let me investigate what xtype isn't getting loaded - thanks for the insight.

Reply:
what about commenting about the beforerender listener? does it still break?

Reply:
Looks like it (all the fields across rows are blank)
![image](https://user-images.githubusercontent.com/10765266/231906482-2bdd90d2-bf9f-4a34-a336-7130a1addd7e.png)

Reply:
"The way this component is supposed to work is that if you have 3 rows selected, it should find all instances where 100% of the rows have the same value. For any field where this is true, it should pre-populate the field."

So, it is currently not working when all the fields are null across rows (100% of the rows with the same null values).

It also does not work in this case...
![image](https://user-images.githubusercontent.com/10765266/232111611-4b09c880-308a-4df1-9a6e-4beda2cf93bb.png)

..unless, the Date field is populated - it looks like that is the requirement from how its currently behaving, and that Date field contains both date and time, which has to be the same across rows; but that is the problem, datetime is not always the same since time can be slightly different, hence it crashes.

Reply:
I dont suppose LabKey has a demo instance of EHR anywhere that's viewable? I believe you are seeing what you're seeing, but I am skeptical of the root cause.  

Did you see my comment about removing the beforerender listener? The sort of behavior you're seeing with a window dying usually happens with a buried JS error somewhere in the code. For example, I believe the data combinations you point out are putting it into some state that throws an error, but I dont believe solutions like grabbing the first value is going to be the right fix.

Is this specific to one form type, or a more general thing?

Reply:
I'm interested in the render listener because of this kind of thing:
```
var cols = win.down('#formPanel').items.get(0).items.getCount();
```

Let's say there's something about this combination of inputs that left "items" empty. Then items.get(0) throws an exception, which could propagate weirdly through Ext4's rendering. I dont know that this is your problem, but this kind of thing is what I often find causing the mode of failure you have here

Reply:
Yes, I looked at that as well, however, commenting out beforerenderer did not make any difference.